src/main/java/graphql/annotations/directives/DirectiveWirer.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.directives;
+
+import graphql.introspection.Introspection;
+import graphql.schema.*;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DirectiveWirer {
+    public GraphQLDirectiveContainer wire(GraphQLDirectiveContainer element, HashMap<GraphQLDirective, AnnotationsDirectiveWiring> directiveWiringMap) {
+        for (Map.Entry<GraphQLDirective, AnnotationsDirectiveWiring> entry : directiveWiringMap.entrySet()) {
+            System.out.println(entry.getKey());
+            GraphQLDirective graphQLDirective = entry.getKey();
+            AnnotationsDirectiveWiring wiring = entry.getValue();
+
+            if (element instanceof GraphQLFieldDefinition) {
```

Comment:
Change to Map<Class, Function> which the key is "GraphQLFieldDefinition,.class" and so on, and the value is the function inside the if statement